### PR TITLE
fix: prioritize current site setting for org value

### PR DESCRIPTION
**Pull request by navinkarkera**
## Description

If multiple tenant configs are enabled for a given org, fetching value from site config for the organization should return current site config value if the org is included in the org_filter.

`Private-ref`: [BB-7927](https://tasks.opencraft.com/browse/BB-7927)

## Testing instructions

See https://github.com/open-craft/eox-tenant/pull/3

## Checklist for Merge

- [ ] Tested in a remote environment
- [x] Updated documentation
- [x] Rebased master/main
- [x] Squashed commits

### CODE REVIEW

**Review by kaustavb12 — APPROVED** [review]
:+1:

- [x] I tested this: Verified that the changes here are same as in the internal PR open-craft/eox-tenant#3
- [x] I read through the code
- [ ] I checked for accessibility issues
- [x] Includes documentation
- [ ] I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository.

**Comment by kaustavb12** [comment]
@navinkarkera Could you please add our internal ticket number here too as private ref, for easy reference.

**Comment by bra-i-am** [comment]
Hello @navinkarkera and @kaustavb12, I hope you are doing well. We thank you for contributing to this project and apologize for the delay in reviewing it due to some urgent matters. 

The solution seems logical. However, before moving forward, I would like to test it to gain a more comprehensive understanding of the expected behavior and the match with all the use cases. 

I followed the steps in the referenced PR https://github.com/open-craft/eox-tenant/pull/3, but it didn't work for me. In order to be more specific, here are the steps I followed:

1. Ran `make {lms,cms}-up`
2. Cloned your repo fork (to be allowed to use the branch `navin/upstream/current-site-priority-for-org-value`), went into the shells in the LMS & CMS, and ran the `pip install -e` command to the repo
3. Ran the two migrations despite only the LMS executed the changes (expected behavior) 
4. Created the two Routes with their corresponding tenant configs the same as you did
5. Updated the redirect URLs in the`studio-sso` application
6. Added the setting in the private.py files
7. Reseted

This is what I receive when I try to see the link when `View live` is hovered:
PR SCREEN

These are my tenant configs with the settings you told in open-craft#3 and there are shown the domains I use too
image

To test this PR I mounted a new environment in `devstack`, please let me know if maybe you have any other setting/configuration I'm missing or maybe you have an idea about what can I do to run this PR properly.

Perhaps, if you have tested this PR using Tutor, could you please guide us a little through the process you followed? 

I will be closely monitoring your response to continue with my review or provide any more information if needed.

Thank you so much ✨

File: eox_tenant/tenant_wise/proxies.py [hunk 1]
```diff
@@ -148,18 +148,26 @@ def __get_value_for_org(cls, org, val_name, default=None):
...
+        tenat_key = getattr(settings, "EDNX_TENANT_KEY", None)
+        if tenat_key is not None:
+            cache_key = "{}-{}".format(cache_key, tenat_key)
         cached_value = cache.get(cache_key)
 
         if cached_value:
             return cached_value
 
-        result = TenantConfig.get_value_for_org(org, val_name)
+        result = TenantConfig.get_value_for_org(org, val_name, tenat_key)
```
Comment (bra-i-am) [inline eox_tenant/tenant_wise/proxies.py:164]:
```suggestion
        result = TenantConfig.get_value_for_org(org, val_name, tenant_key)
```

File: eox_tenant/tenant_wise/proxies.py [hunk 2]
```diff
@@ -148,18 +148,26 @@ def __get_value_for_org(cls, org, val_name, default=None):
...
 
         if result:
             cls.set_key_to_cache(cache_key, result)
             return result
 
-        result = Microsite.get_value_for_org(org, val_name)
+        result = Microsite.get_value_for_org(org, val_name, tenat_key)
```
Comment (bra-i-am) [inline eox_tenant/tenant_wise/proxies.py:170]:
```suggestion
        result = Microsite.get_value_for_org(org, val_name, tenant_key)
```

File: eox_tenant/tenant_wise/proxies.py [hunk 3]
```diff
@@ -148,18 +148,26 @@ def __get_value_for_org(cls, org, val_name, default=None):
...
         """
         cache_key = "org-value-{}-{}".format(org, val_name)
+
+        # Make use of tenant-external-key to generate unique cache_key per
+        # tenant. This will help to fetch the current tenant value if the org
+        # is configured in multiple tenants/microsites including the current
+        # one.
+        tenat_key = getattr(settings, "EDNX_TENANT_KEY", None)
+        if tenat_key is not None:
+            cache_key = "{}-{}".format(cache_key, tenat_key)
```
Comment (bra-i-am) [inline eox_tenant/tenant_wise/proxies.py:158]:
```suggestion
        tenant_key = getattr(settings, "EDNX_TENANT_KEY", None)
        if tenant_key is not None:
            cache_key = "{}-{}".format(cache_key, tenant_key)
```

**Comment by navinkarkera** [comment]
Hello @bra-i-am, I just rebased the branch with master and followed the steps mentioned in https://github.com/open-craft/eox-tenant/pull/3 to test in master devstack. It worked for me and I did not do anything special.

lacolhost-screenshot
localhost-screenshot


Can you please make sure that you are using open-craft:navin/upstream/current-site-priority-for-org-value branch?

**Review by bra-i-am — APPROVED** [review]
@navinkarkera, thank you so much for your answer. I could not reproduce the behavior before because of some config mistakes I made with my routes, but I finally got to solve them.

The change introduced in this PR makes all the sense to me and according to some tests I ran, there are no conflicts with other test cases. I'll announce this to my teammates to continue with the merge.

Thanks a lot for your contribution ✨

### Evidence
image


### =====================================

#### **FYI**
- I ran the PR using Tutor in dev mode changing the variable `CMS_HOST` same as `LMS_HOST` in the `config.yml` file (e.g. `CMS_HOST: local.overhang.io`)
- I created the routes `tenant-a.local.overhang.io` and `tenant-b.local.overhang.io`
- I created the Tenant Configs according to the shown [here](https://github.com/open-craft/eox-tenant/pull/3#:~:text=Add%20an%20entry%20with%20below%20details%3A) but I changed the URL to the one of the route that is being configured. For example the `lms_configs` for the Tenant A:

```yml
{
  "EDNX_USE_SIGNAL": true,
  "LMS_BASE": "tenant-a.local.overhang.io:8000",
  "LMS_ROOT_URL": "http://tenant-a.local.overhang.io:8000",
  "SOCIAL_AUTH_EDX_OAUTH2_PUBLIC_URL_ROOT": "http://tenant-a.local.overhang.io:8000",
  "course_org_filter": [
    "edX"
  ]
}
```

**Review by MaferMazu — APPROVED** [review]
Looks good to me. I will change the commit to fix because we are changing the previous behavior.

**Comment by MaferMazu** [comment]
@navinkarkera, can you sign your commit, please?

Ref: https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification

**Comment by navinkarkera** [comment]
@MaferMazu Done.

**Pull request merged**